src/Post/Post.js

```diff
@@ -359,6 +364,85 @@ function CommentSection({
     </section>
   );
 }
+function MediaSection({ media_items, user_name }) {
+  const [mediaIndex, setMediaIndex] = React.useState(0);
+  const [videoPause, setVideoPause] = React.useState(false);
+  return (
+    <section className="img-container">
+      {media_items[mediaIndex].type === 'photo' ? (
+        <img
+          className="post-img"
+          src={media_items[mediaIndex].url}
+          alt={`${user_name}  Avatar`}
+          loading="lazy"
+        />
+      ) : (
+        <>
+          {videoPause && (
+            <div className="play-icon-container">
+              <PlayArrowRounded
+                onClick={() => setVideoPause(false)}
+                className="play-icon"
+                fontSize="large"
+              />
+            </div>
+          )}
+
+          <div
+            onClick={() => setVideoPause(!videoPause)}
+            className="post-video-container"
+          >
+            <Video
+              videoURL={media_items[mediaIndex].url}
+              paused={videoPause}
+              className="post-video"
+              autoPlay={false}
+            />
```

Comment:
This is good idea but only for a shor time. I'd create another component called `PostVideo` and leave this untouched. Because `StoryVideo` does some time stuff and tracks the progress and so on, and has no Play button. And videos in the post have play buttons. Also the stories have fixed aspect ratio (tall and thin), while post videos come in all sizes.

Reply:
Awesome suggestion.
Getting it done.